src/libraries/Microsoft.Extensions.HostFactoryResolver/src/HostFactoryResolver.cs

```diff
@@ -163,9 +163,9 @@ static bool IsApplicationNameArg(string arg)
                         => arg.Equals("--applicationName", StringComparison.OrdinalIgnoreCase) ||
                             arg.Equals("/applicationName", StringComparison.OrdinalIgnoreCase);
 
-                    args = args.Any(arg => IsApplicationNameArg(arg)) || assembly?.GetName().Name is null
+                    args = (args.Any(arg => IsApplicationNameArg(arg)) || assembly is null || assembly.GetName().Name is null
```

Comment:
What is the impactful difference here?

Reply:
But the code being changed is `assembly?.GetName().Name is null` which is the same meaning as the `assembly is null || assembly.GetName().Name is null` this PR is changing it to. What am I missing?